app/uk/gov/hmrc/agentuserclientdetails/controllers/ClientListController.scala

```diff
@@ -97,7 +101,9 @@ class ClientListController @Inject() (
         }
     }
 
-  protected def getClientsForGroupIdFn(groupId: String)(implicit request: RequestHeader): Future[Result] = {
+  protected def getClientsForGroupIdFn(
+    groupId: String
+  )(implicit maybeArn: Option[Arn], request: RequestHeader): Future[Result] = {
     def makeWorkItem(client: Client)(implicit hc: HeaderCarrier): FriendlyNameWorkItem = {
```

Comment:
This function would fail to create job tracking data if an Arn was not supplied (None), and would not warn us or log anything in that case. I think the Arn should not be optional.
Also it should be in the main argument list rather than as an implicit.
As for how to handle the case when the controller is called with a groupId rather than an Arn, we have the choice to either:

1. Abolish the 'group id' endpoint (when I first started writing AUCD it seemed we would use groupId more than ARN but it turns out to be the opposite now). But we must check if any other service uses the 'groupId' endpoint first!
2. Retrieve the Arn for that groupId via ES3.